FiftyOne.DeviceDetection/DeviceDetectionOnPremisePipelineBuilder.cs

```diff
@@ -404,6 +410,111 @@ public DeviceDetectionOnPremisePipelineBuilder SetPerformanceProfile(Performance
             return this;
         }
 
+        /// <summary>
+        /// Configure the engine to use the specified URL when looking for
+        /// an updated data file.
+        /// </summary>
+        /// <param name="url">
+        /// The URL to check for a new data file.
+        /// </param>
+        /// <returns>
+        /// This builder instance.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown if url parameter is null or and empty string
+        /// </exception>
+        public DeviceDetectionOnPremisePipelineBuilder SetDataUpdateUrl(string url)
+        {
+            if (string.IsNullOrEmpty(url)) throw new ArgumentNullException(nameof(url));
+            _dataUpdateUrlString = url;
+            _dataUpdateUrlUri = null;
+            return this;
+        }
+
+        /// <summary>
+        /// Configure the engine to use the specified URL when looking for
+        /// an updated data file.
+        /// </summary>
+        /// <param name="url">
+        /// The URL to check for a new data file.
+        /// </param>
+        /// <returns>
+        /// This builder instance.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown if url parameter is null
+        /// </exception>
+        public DeviceDetectionOnPremisePipelineBuilder SetDataUpdateUrl(Uri url)
+        {
+            if (url == null) throw new ArgumentNullException(nameof(url));
+            _dataUpdateUrlString = null;
+            _dataUpdateUrlUri = url;
+            return this;
+        }
+
+        /// <summary>
+        /// Specify a <see cref="IDataUpdateUrlFormatter"/> to be 
+        /// used by the <see cref="DataUpdateService"/> when building the 
+        /// complete URL to query for updated data.
+        /// </summary>
+        /// <param name="formatter">
+        /// The formatter to use.
+        /// </param>
+        /// <returns>
+        /// This builder instance.
+        /// </returns>
+        public DeviceDetectionOnPremisePipelineBuilder SetDataUpdateUrlFormatter(
+            IDataUpdateUrlFormatter formatter)
+        {
+            _dataUpdateUrlFormatter = formatter;
+            _useFormatter = null;
```

Comment:
do we need to set `_useFormatter=null`, or can `_dataUpdateUrlFormatter` and `_useFormatter` be completely independent?